othello/apps/games/consumers.py

```diff
@@ -0,0 +1,106 @@
+from asgiref.sync import async_to_sync
+from channels.generic.websocket import JsonWebsocketConsumer
+
+from django.utils import timezone
+
+from ...moderator.moderator import Player
+from .models import Game, Submission
+from .tasks import run_game
+from .utils import serialize_game_error, serialize_game_info, serialize_game_log
+
+
+class GameConsumer(JsonWebsocketConsumer):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.game = None
+        self.connected = False
+
+    def connect(self):
+        try:
+            self.game = Game.objects.get(id=self.scope["url_route"]["kwargs"]["game_id"])
+        except Game.DoesNotExist:
+            self.close()
+            return
+
+        yourself = Submission.objects.get(user__username="Yourself")
+
+        if not self.game.playing:
+            self.close()
+            return
+
+        self.is_black_yourself = self.game.black == yourself
+        self.is_white_yourself = self.game.white == yourself
+
+        self.connected = True
+        self.accept()
+
+        async_to_sync(self.channel_layer.group_add)(
+            self.game.channels_group_name, self.channel_name
+        )
+
+    def disconnect(self, code):
+        self.connected = False
+        super().disconnect(code=code)
+        self.close()
```

Comment:
No, no, no. `disconnect()` is called automatically when the websocket is disconnected. Remove the `close()` call and call `disconnect()` elswehere.

Also, the default `disconnect()` does nothing -- it's just there for you to override if you want. You don't need to call `super().disconnect()`.